gpytorch/kernels/rbf_kernel.py

```diff
@@ -73,7 +73,7 @@ def forward(self, x1, x2, diag=False, **params):
         if (
             x1.requires_grad
             or x2.requires_grad
-            or (self.ard_num_dims is not None and self.ard_num_dims > 1)
+            or self.lengthscale.numel() > 1
```

Comment:
This is not equivalent if `batch_shape` is not set. Is this intended?

Reply:
This will probably get reverted in the clean up -- the Kernel will have to override `pyro_load_from_samples` anyways to update batch_shape. Right now, if batch_shape is wrong, you can only forward through the model with debug mode off.